core/server/master/src/main/java/alluxio/master/file/replication/ReplicationChecker.java

```diff
@@ -220,12 +202,6 @@ public void close() {
   private void checkMisreplicated(Set<Long> inodes, ReplicationHandler handler)
       throws InterruptedException {
     for (long inodeId : inodes) {
-      if (mActiveJobToInodeID.size() >= mMaxActiveJobs) {
-        return;
-      }
-      if (mActiveJobToInodeID.containsValue(inodeId)) {
```

Comment:
don't we check duplicated job any more?

Reply:
Unfortunately we.do need to. And given the current job master interface, that means checking each job for completion status to maintain the active job list. I will try to see if we can make a smaller change.here without changing job master interface, but that may not be possible